module/os/freebsd/spl/spl_taskq.c

```diff
@@ -143,8 +164,10 @@ taskq_lookup(taskqid_t tqid)
 static taskqid_t
 taskq_insert(taskq_ent_t *ent)
 {
-	taskqid_t tqid = atomic_fetchadd_int(&tqidnext, 1);
+	taskqid_t tqid;
 
+	tqid = __taskq_genid();
+	VERIFY(tqid);
```

Comment:
An `ASSERT` at the end of `__taskq_genid()` seems sufficient for this and would cover all callers.

Reply:
Well there are two and merging them into one is imo uglier. Still, pushed asserts into them.